app/helpers/application_helper/navbar.rb

```diff
@@ -19,7 +19,7 @@ def item_to_hash(item)
         :type    => item.type,
         :visible => item.visible?,
         :active  => item_active?(item),
-        :items   => item.items.to_a.map(&method(:item_to_hash))
+        :items   => item.items.to_a.select { |i| i.visible? }.map(&method(:item_to_hash))
```

Comment:
I think this is a regression as this has been working before the react menus. Maybe we should address it there.

Reply:
@skateman first level menu item is being taken care of at https://github.com/ManageIQ/manageiq-ui-classic/blob/b104aada452b1f690b3d5f18ba9b1a20aa6d2ae6/app/helpers/application_helper/navbar.rb#L6 so i am not sure why we should hide second level item in the react code. In my opinion this fix should stay.

Reply:
Okay, I'm convinced :laughing: but the select should be simplified:
```suggestion
        :items   => item.items.to_a.select(&:visible).map(&method(:item_to_hash))
```